### PR TITLE
Don't log error level logs for non critical errors

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

File: IdentityCore/src/oauth2/MSIDOauth2Factory.m [hunk 1]
```diff
@@ -229,7 +229,7 @@ - (BOOL)fillAccessToken:(MSIDAccessToken *)token
     
     if (!token.accessToken)
     {
-        MSID_LOG_ERROR(nil, @"Trying to initialize access token when missing access token field");
+        MSID_LOG_WARN(nil, @"Trying to initialize access token when missing access token field");
```
Comment (antrix1989) [inline IdentityCore/src/oauth2/MSIDOauth2Factory.m]: Why this is "non-critical"? Don't we require AT to be present in server response?

File: IdentityCore/src/oauth2/token/MSIDPrimaryRefreshToken.m [hunk 2]
```diff
@@ -40,7 +40,7 @@ - (instancetype)initWithTokenCacheItem:(MSIDCredentialCacheItem *)tokenCacheItem
         
         if (!_sessionKey)
         {
-            MSID_LOG_ERROR(nil, @"Trying to initialize primary refresh token when missing session key field");
+            MSID_LOG_WARN(nil, @"Trying to initialize primary refresh token when missing session key field");
```
Comment (antrix1989) [inline IdentityCore/src/oauth2/token/MSIDPrimaryRefreshToken.m]: Isn't this a critical error?

Comment (oldalton) [reply IdentityCore/src/oauth2/token/MSIDPrimaryRefreshToken.m]: Corrupted items in keychain are skipped and considered non critical.

File: IdentityCore/src/oauth2/MSIDOauth2Factory.m (hunk 1 above)
Comment (oldalton) [reply IdentityCore/src/oauth2/MSIDOauth2Factory.m]: Fixed this one. You're right, this one is not reading it from cache, so should be error level.

**Pull request merged**